File: BACKEND/src/weylo.admin.api/Services/AdminService.cs
using Microsoft.EntityFrameworkCore;
using weylo.admin.api.Services.Interfaces;
using weylo.shared.Data;
using weylo.shared.Models;

namespace weylo.admin.api.Services
{
    public class AdminService : IAdminService
    {
        private readonly BaseDbContext _context;

        public AdminService(BaseDbContext context)
        {
            _context = context;
        }

        public async Task<List<User>> GetUsersAsync()
        {
            return await _context.Users
                .OrderBy(u => u.Username)
                .ToListAsync();
        }

        public async Task<bool> DeleteUserAsync(int userId)
        {
            var user = await _context.Users.FindAsync(userId);
            if (user == null)
                return false;

            _context.Users.Remove(user);
            await _context.SaveChangesAsync();
            return true;
        }

        public async Task<bool> ChangeUserRoleAsync(int userId, string newRole)
        {
            var user = await _context.Users.FindAsync(userId);
            if (user == null)
                return false;

            user.Role = newRole;
            user.UpdatedAt = DateTime.UtcNow;

            await _context.SaveChangesAsync();
            return true;
        }
        public async Task<User?> GetUserByIdAsync(int userId)
        {
            return await _context.Users.FindAsync(userId);
        }
    }
}

File: FRONTEND/weylo.front/src/app/services/AdminService.ts
import { ApiResponse, User } from "../types/shared";
import httpClient from "./httpClient";

class AdminService {
  async getUsers(): Promise<ApiResponse<User[]>> {
    return httpClient.get<User[]>("/api/admin/users");
  }

  async deleteUser(userId: number): Promise<ApiResponse<{ message: string }>> {
    return httpClient.delete<{ message: string }>(`/api/admin/users/${userId}`);
  }

  async changeUserRole(
    userId: number,
    newRole: string
  ): Promise<ApiResponse<{ message: string }>> {
    return httpClient.put<{ message: string }>(
      `/api/admin/users/${userId}/role`,
      { newRole }
    );
  }
}

const adminService = new AdminService();
export default adminService;
